src/ophys_etl/pipelines/segment_binarize_pipeline.py

```diff
@@ -1,54 +1,96 @@
+import json
+from pathlib import Path
+import tempfile
+
+import marshmallow
 import argschema
-from ophys_etl.transforms.suite2p_wrapper import Suite2PWrapper
-from ophys_etl.transforms.convert_rois import BinarizerAndROICreator
-from ophys_etl.schemas.fields import H5InputFile
 
-import tempfile
-import pathlib
-import json
+
+from ophys_etl.transforms.convert_rois import (
+    BinarizeAndCreateROIsInputSchema, BinarizerAndROICreator)
+from ophys_etl.transforms.suite2p_wrapper import (Suite2PWrapper,
+                                                  Suite2PWrapperSchema)
+
+
+class ConvertROIsInputSchema(BinarizeAndCreateROIsInputSchema):
+
+    # Override the default "suite2p_stat_field" required parameter of the
+    # BinarizeAndCreateROIsInputSchema.
+    # When run in pipeline, the suite2p_stat_path won't be known until
+    # after segmentation has run.
+    suite2p_stat_path = argschema.fields.Str(
+        required=False,
+        validate=lambda x: Path(x).exists(),
+        description=("Path to s2p output stat file containing ROIs generated "
+                     "during source extraction"))
 
 
 class SegmentBinarizeSchema(argschema.ArgSchema):
-    motion_corrected_video = H5InputFile(
-        required=True,
-        description="hdf5 motion-correced video")
-    motion_correction_values = argschema.fields.InputFile(
-        required=True,
-        description="motion correction values in a csv")
+    suite2p_args = argschema.fields.Nested(Suite2PWrapperSchema,
+                                           required=True)
+    convert_args = argschema.fields.Nested(ConvertROIsInputSchema,
+                                           required=True)
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.tmpdir = None
+
+    @marshmallow.pre_load
+    def setup_default_suite2p_args(self, data: dict, **kwargs) -> dict:
+
+        # suite2p_args
+        if "output_dir" not in data["suite2p_args"]:
+            self.tmpdir = tempfile.TemporaryDirectory()
+            data["suite2p_args"]["output_dir"] = self.tmpdir.name
+
+        if "output_json" not in data["suite2p_args"]:
+            Suite2p_output = (Path(data["suite2p_args"]["output_dir"])
+                              / "Suite2P_output.json")
+            data["suite2p_args"]["output_json"] = str(Suite2p_output)
+
+        if "bin_size" not in data["suite2p_args"]:
+            data["suite2p_args"]["bin_size"] = 115
```

Comment:
I think for the time-being, this should raise a warning that this default is being applied. Perhaps:
```
warnings.warn("argument 'bin_size' was not specified. A default value of 115 is being used. "
                        "This setting is probably appropriate for 31Hz frame rates, but not necessarily for "
                        "others.")
```

Reply:
Note to myself: This will not be an issue once #81 is merged as it gets rid of the `bin_size` arg completely.